context/app/static/js/components/workspaces/WorkspacesList.jsx

```diff
@@ -1,25 +1,39 @@
-import React from 'react';
+import React, { useRef } from 'react';
 import Paper from '@mui/material/Paper';
 import Typography from '@mui/material/Typography';
+import Stack from '@mui/material/Stack';
+import DeleteRounded from '@mui/icons-material/DeleteRounded';
+import AddRounded from '@mui/icons-material/AddRounded';
 
 import Description from 'js/shared-styles/sections/Description';
-import { DeleteIcon } from 'js/shared-styles/icons';
 import { SpacedSectionButtonRow } from 'js/shared-styles/sections/SectionButtonRow';
-import { PanelWrapper } from 'js/shared-styles/panels';
 
-import WorkspaceDetails from 'js/components/workspaces/WorkspaceDetails';
-import CreateWorkspaceButton from 'js/components/workspaces/CreateWorkspaceButton';
+import WorkspaceListItem from 'js/components/workspaces/WorkspaceListItem';
+import { useSelectItems } from 'js/hooks/useSelectItems';
+import { useSnackbarActions } from 'js/shared-styles/snackbars';
 import { useWorkspacesList } from './hooks';
-import { StyledButton } from './style';
+import WorkspaceButton from './WorkspaceButton';
 
 function WorkspacesList() {
-  const {
-    workspacesList,
-    handleDeleteWorkspace: handleDelete,
-    handleCreateWorkspace,
-    handleStopWorkspace: handleStop,
-    handleStartWorkspace,
-  } = useWorkspacesList();
+  const { workspacesList, handleCreateWorkspace, handleDeleteWorkspace } = useWorkspacesList();
+
+  const { selectedItems, toggleItem } = useSelectItems();
+  const { toastError } = useSnackbarActions();
+
+  const isDeleting = useRef(false);
```

Comment:
Perhaps I'm mistaken, but updating the ref doesn't cause a re-render and the button wouldn't show to be disabled while `isDeleting` is true?

Reply:
This should now be handled appropriately by updating all workspaces list actions to use `useSWRMutation`, which lets us use the loading state provided by the hook.